internal/controller/resource_manager.go

```diff
@@ -102,7 +102,15 @@ func (r *ResourceManager) DynamicListResources(ctx context.Context, resource str
 	}
 	for _, obj := range objs {
 		metadataObj := convertMetaObjectToMetadata(obj, resource)
-		err := sendObjectMetaData(r.streamManager, metadataObj)
+		if resource == "pods" {
+			hostIPs, err := getPodIPAddresses(ctx, r.logger, obj.GetName(), obj.GetNamespace())
```

Comment:
`resource` should already contains that data. Don't re-query it.
Modify `convertMetaObjectToMetadata` to add the kind-specific data into the returned protobuf.
And rename `convertMetaObjectToMetadata` into `convertMetaObjectToInventoryData` or something like that to clarify its expanded scope.

Reply:
The meta point here was to move the logic into `convertMetaObjectToMetadata` so you don't have copy-and-paste the `if`/`switch` below multiple times.

Reply:
Ohh yeah I get what you are saying! Will do.